### PR TITLE
Task composer restructure

### PR DESCRIPTION
**Pull request by Levi-Armstrong**
Task Composer is configurable using plugins and the restructure is similar to other Tesseract package which are plugin based. This PR separates out the core components which only depend on tesseract_common and system dependencies to allow this framework to be used for things like tool path filtering, perception pipelines, etc.

### CODE REVIEW

**Comment by marip8** [comment]
Can you add some description of the changes in this PR or a diagram that shows the architecture? It would be nice to understand the overall goals before reviewing

File: tesseract_task_composer/cmake/tesseract_task_composer-config.cmake.in [hunk 1]
```diff
@@ -0,0 +1,39 @@
+@PACKAGE_INIT@
```
Comment (marip8) [inline tesseract_task_composer/cmake/tesseract_task_composer-config.cmake.in:1]: Doesn't the current version of RICB take care of this without having to create a separate file?

Comment (Levi-Armstrong) [reply tesseract_task_composer/cmake/tesseract_task_composer-config.cmake.in:1]: I just followed how things were done in tesseract_kinematics, though I am not sure it supports conditional dependencies.

**Comment by Levi-Armstrong** [comment]
> Can you add some description of the changes in this PR or a diagram that shows the architecture? It would be nice to understand the overall goals before reviewing

I updated the description in the PR.

**Comment by Levi-Armstrong** [comment]
The main difference change is the addition of PlanningTaskComposerProblem and TaskComposerProblem contains minimal object required for core functionality which is the first commit. The second commit is the restructure to separate dependencies.

**Pull request merged**